### PR TITLE
Assign default roles from Teams to User

### PR DESCRIPTION
**Pull request by mithmatt**
### Describe your changes :

See #2883

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Backend: @sureshms @harshach

### CODE REVIEW

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java [hunk 1]
```diff
@@ -109,20 +109,49 @@ void post_userWithDefaultRole(TestInfo test) throws IOException {
...
+
+    // Given team1 with role1 set as default, when a user is created with team1, then the user should have the global
+    // default role and role1 assigned.
+    CreateUser create = createRequest(test, 1).withTeams(List.of(team1.getId()));
+    createUserAndCheckRoles(create, Arrays.asList(defaultRoleId, role1.getId()));
+
+    // Given team1 with role1 set as default and team2 with role1 and role2 set as default, when a user is created
+    // with team1 and team2, then the user should have the global default role, role1 and role2 assigned.
+    create = createRequest(test, 2).withTeams(List.of(team1.getId(), team2.getId()));
+    createUserAndCheckRoles(create, Arrays.asList(defaultRoleId, role1.getId(), role2.getId()));
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java:152]: Should probably add a test to showcase removal of a role when user changes teams

**Comment by mithmatt** [comment]
pytest failure due to connection issue

```
Run sudo apt-get install -y libsasl2-dev unixodbc-dev python3-venv
Reading package lists...
Building dependency tree...
Reading state information...
python3-venv is already the newest version (3.[8](https://github.com/open-metadata/OpenMetadata/runs/5315142730?check_suite_focus=true#step:5:8).2-0ubuntu2).
unixodbc-dev is already the newest version (2.3.7).
The following NEW packages will be installed:
  libsasl2-dev
0 upgraded, 1 newly installed, 0 to remove and 33 not upgraded.
Need to get 225 kB of archives.
After this operation, 8[9](https://github.com/open-metadata/OpenMetadata/runs/5315142730?check_suite_focus=true#step:5:9)4 kB of additional disk space will be used.
Err:1 http://azure.archive.ubuntu.com/ubuntu focal/main amd64 libsasl2-dev amd64 2.1.27+dfsg-2
  Could not connect to azure.archive.ubuntu.com:80 (52.[14](https://github.com/open-metadata/OpenMetadata/runs/5315142730?check_suite_focus=true#step:5:14)7.219.192), connection timed out
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/c/cyrus-sasl2/libsasl2-dev_2.1.27+dfsg-2_amd64.deb  Could not connect to azure.archive.ubuntu.com:80 (52.147.2[19](https://github.com/open-metadata/OpenMetadata/runs/5315142730?check_suite_focus=true#step:5:19).192), connection timed out
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

**Review by pmbrull — APPROVED** [review]

**Comment by vivekratnavel** [comment]
Hi @mithmatt,

Thanks for the PR. Can you please take care of this test failure?
```
Errors: 
2022-02-24T07:06:04.7753691Z [ERROR]   UserResourceTest.userDefaultRoleAssignment:126->EntityResourceTest.patchEntityAndCheck:1480->EntityResourceTest.patchEntityAndCheck:1505->EntityResourceTest.patchEntity:1321 » HttpResponse 
```

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/RoleResourceTest.java [hunk 2]
```diff
@@ -78,25 +79,19 @@ void defaultRole(TestInfo test) throws IOException {
 
       List<User> users = new UserResourceTest().listEntities(Map.of("fields", "roles"), ADMIN_AUTH_HEADERS).getData();
       for (User user : users) {
-        boolean defaultRoleSet = false, prevDefaultRoleExists = false;
-
-        for (EntityReference role : user.getRoles()) {
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/RoleResourceTest.java:83]: Simplify the code - make it easy to understand

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java [hunk 3]
```diff
@@ -100,27 +101,90 @@ public void post_entity_as_non_admin_401(TestInfo test) {
 
   @Order(Integer.MAX_VALUE) // Run this test last to avoid side effects of default role creation to fail other tests.
   @Test
-  void post_userWithDefaultRole(TestInfo test) throws IOException {
+  void userDefaultRoleAssignment(TestInfo test) throws IOException {
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/UserResourceTest.java:104]: Making this a comprehensive behavioral test

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java [hunk 4]
```diff
@@ -66,24 +66,29 @@ public UserRepository(CollectionDAO dao) {
...
-    Set<UUID> defaultRoleIds =
-        daoCollection.roleDAO().getDefaultRolesIds().stream().map(UUID::fromString).collect(Collectors.toSet());
-    defaultRoleIds.removeAll(existingRoleIds);
+    // Add default role set up globally.
+    daoCollection.roleDAO().getDefaultRolesIds().forEach(roleIdStr -> roleIds.add(UUID.fromString(roleIdStr)));
 
-    if (rolesRef == null || rolesRef.isEmpty()) {
-      rolesRef = new ArrayList<>();
+    // Add default roles from the teams that the user belongs to.
+    List<EntityReference> teamsRef = Optional.ofNullable(user.getTeams()).orElse(Collections.emptyList());
```
Comment (mithmatt) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java:79]: This is the new logic being added

**Pull request merged**